fs/fs.go

```diff
@@ -29,7 +29,7 @@ func NewDatastore(path string) (ds.Datastore, error) {
 
 // KeyFilename returns the filename associated with `key`
 func (d *Datastore) KeyFilename(key ds.Key) string {
-	return filepath.Join(d.path, key.String(), ObjectKeySuffix)
```

Comment:
yeah, hiding objects is a hack. :/  -- giving dirs ext may be ok. 

Either way-- if think it'll impact perf significantly let's keep it outside. 

We still have to give dirs extensions (or some other workaround) because: 

```
ds.Put("/foo", "foo")
ds.Put("/foo.object/bar", "ha!")
```
:(